packages/xo-server/src/xapi/mixins/vm.mjs

```diff
@@ -213,6 +213,10 @@ const methods = {
       )
     }
 
+    if (vm.platform.vtpm === 'true') {
+      await this.VTPM_create({ VM: vm.$ref })
+    }
+
```

Comment:
What about the existing widget in the VM creation page's advanced settings? It looks like this bypasses it. So users creating a new Windows 11 VM and opening the advanced settings before hitting "Create" would see the widget is off, like no vTPM would be created, when in fact a vTPM will still be created.

I also found out something else about how XAPI behaves regarding vTPMs. On XOA latest (so without this commit), I created a VM from the Windows 11 templates, but without enabling vTPM. This created no vTPM, as expected. But as soon as I started the VM, a vTPM was created by XAPI. I believe they did this based on `vm.platform.vtpm`.

Consequently, I believe the best behaviour would be, at VM creation, to make the widget status automatically match the template's `platform.vtpm` field, and change what this widget does: if set to `true`, by users, define `platform.vtpm` as true and create the vTPM (or let XAPI create it at first boot, as you prefer. Creating it avoids the situation where users set it to true but don't start the VM immediately and might notice there's no vTPM yet), and if set to `false` by user, make sure `vm.platform.vtpm` is set to `false` too so that XAPI doesn't create one at first boot.